var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -196,6 +196,10 @@ def cmake_args(self):
         if spec.satisfies('@20180629:+lib'):
             args.append('-DBUILD_LIB=ON')
 
+        if spec.satisfies('%aocc'):
+            cxx_flags = '-Ofast -mfma -fvectorize -funroll-loops'
+            args.append('-DCMAKE_CXX_FLAGS_RELEASE={0}'.format(cxx_flags))
```

Comment:
Generally compiler flags should be defined through the [flag handler](https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=flag_handler#compiler-flags) since they offer better control of what actually goes to the compiler (via the spack compiler wrappers). Will that work here? Something like:
```python
def flag_handler(self, name, flags):
    if name == 'cxxflags' and self.spec.satisfies('%aocc build_type=Release'):
        flags = ['-Ofast', '-mfma', '-fvectorize', '-funroll-loops']
    return (flags, None, None)
```

Reply:
Hi @sethrj , thanks for your comments,
I tried passing flags using flag handler, but its not able to set flags for `-DCMAKE_CXX_FLAGS_RELEASE` , using flag handler we able to set `cflags`, `cxxflags` and `cppflags` but not able to set `CXX_FLAGS_RELEASE`, hence it is written like above

Reply:
So is the problem that `CMAKE_CXX_FLAGS_RELEASE` defines existing flags that you're not overriding? I'm OK with this then but would recommend using the package helper macros:
```suggestion
            args.append(self.define('CMAKE_CXX_FLAGS_RELEASE, cxx_flags))
```

Reply:
Addressed this in commit ID: e1e2289